### PR TITLE
check if the build branch type is coherent between what the user has specified and what is entered in the changelog

### PR DESCRIPTION
**Pull request by OniriCorpe**
because it happens lol

idk if i put the code in the right place

### CODE REVIEW

File: ynh-build [hunk 1]
```diff
@@ -82,6 +82,12 @@ function checkout_tag()
 
 function build()
 {
+   # check if the build branch type is coherent between what the user has specified and what is entered in the changelog
+   if ! [[ head -n 1 debian/changelog == *"$BRANCH"* ]]; then
```
Comment (alexAubin) [inline ynh-build]:
```suggestion
   if ! [[ "$(head -n 1 debian/changelog)" == *"$BRANCH"* ]]; then
```

File: ynh-build [hunk 2]
```diff
@@ -82,6 +82,12 @@ function checkout_tag()
...
+      echo "The specified branch type is not coherent with the changelog one!"
```
Comment (alexAubin) [inline ynh-build:87]:
```suggestion
      echo "The specified branch type is not consistent with the changelog one!"
```

File: ynh-build [hunk 3]
```diff
@@ -82,6 +82,12 @@ function checkout_tag()
 
 function build()
 {
+   # check if the build branch type is coherent between what the user has specified and what is entered in the changelog
+   if ! [[ $(head -n 1 debian/changelog == *"$BRANCH"*) ]]; then
```
Comment (alexAubin) [inline ynh-build]:
```suggestion
   if ! [[ "$(head -n 1 debian/changelog)" == *"$BRANCH"* ]]; then
```

:grimacing:

File: ynh-build (hunk 1 above)
Comment (OniriCorpe) [reply ynh-build]: yeah i've done this on my tests but on the file x)

File: ynh-build (hunk 3 above)
Comment (OniriCorpe) [reply ynh-build]: sigh i'm tired omg

**Review by alexAubin — APPROVED** [review]

**Pull request merged**